web/js/codeworld.js

```diff
@@ -1261,83 +1251,6 @@ function downloadProject() {
   }
 }
 
-function parseCompileErrors(rawErrors) {
-  const errors = [];
-  rawErrors = rawErrors.split('\n\n');
-  rawErrors.forEach((err) => {
-    const lines = err.trim().split('\n');
-    const firstLine = lines[0].trim();
-    const otherLines = lines
-      .slice(1)
-      .map((ln) => ln.trim())
-      .join('\n');
-    const re1 = /^program\.hs:(\d+):((\d+)-?(\d+)?): (\w+):(.*)/;
-    const re2 = /^program\.hs:\((\d+),(\d+)\)-\((\d+),(\d+)\): (\w+):(.*)/;
-
-    if (err.trim() === '') {
-      // Ignore empty messages.
-    } else if (re1.test(firstLine)) {
-      const match = re1.exec(firstLine);
-
-      const line = Number(match[1]) - 1;
-      let startCol = Number(match[3]) - 1;
-      let endCol;
-      if (match[4]) {
-        endCol = Number(match[4]);
-      } else {
-        const token = window.codeworldEditor
-          .getLineTokens(line)
-          .find((t) => t.start === startCol);
-        if (token) {
-          endCol = token.end;
-        } else if (
-          startCol >= window.codeworldEditor.getDoc().getLine(line).length
-        ) {
-          endCol = startCol;
-          --startCol;
-        } else {
-          endCol = startCol + 1;
-        }
-      }
-
-      const message = ((match[6] ? `${match[6].trim()}\n` : '') + otherLines)
-        .replace(/program\.hs:(\d+):((\d+)(-\d+)?)/g, 'Line $1, Column $2')
-        .replace(
-          /program\.hs:\((\d+),(\d+)\)-\((\d+),(\d+)\)/g,
-          'Line $1-$3, Column $2-$4'
-        );
-
-      errors.push({
-        from: CodeMirror.Pos(line, startCol),
-        to: CodeMirror.Pos(line, endCol),
-        severity: match[5],
-        fullText: err,
-        message: message,
-      });
-    } else if (re2.test(firstLine)) {
-      const match = re2.exec(firstLine);
-
-      const startLine = Number(match[1]) - 1;
-      const startCol = Number(match[2]) - 1;
-      const endLine = Number(match[3]) - 1;
-      const endCol = Number(match[4]);
-
-      errors.push({
-        from: CodeMirror.Pos(startLine, startCol),
-        to: CodeMirror.Pos(endLine, endCol),
-        severity: match[5],
-        fullText: err,
-        message: (match[6] ? `${match[6].trim()}\n` : '') + otherLines,
-      });
-    } else {
-      console.log('Can not parse error header:', firstLine);
-    }
-  });
-  return errors;
-}
-
-function clearWorkspace() {
-  utils.directoryTree.clearSelectedNode();
-
-  setCode('');
-}
+// TEMP: required by setCode in codeworld_shared.js
```

Comment:
Do you want to merge first and then fix this?  Or is this a work in progress?

Reply:
Let's prioritize merging this PR. These are things that I can take care of later (blocks page has few of those globally defined methods as well).